ServiceApp/src/org/droidplanner/services/android/core/gcs/follow/Follow.java

```diff
@@ -103,11 +123,38 @@ public boolean isEnabled() {
         return state == FollowStates.FOLLOW_RUNNING || state == FollowStates.FOLLOW_START;
     }
 
+    public FollowLocationSource getLocationSource() {
+        return mLocationSource;
+    }
+
+    public void setLocationSource(FollowLocationSource source) {
```

Comment:
@kellyschrock if a `FollowLocationSource.INTERNAL` or `FollowLocationSource.NONE` parameter is passed to this method, it'll trigger the `locationFinder` without checking if follow is in an enabled state.
There may also be confusion on how this method is used, since you can pass a `FollowLocationSource` parameter both here, and to the `toggleFollowMeState(...)` method.

Let's proceed with replicating what you did for the `disableFollowMe` method: 
* make the `enableFollowMe(...)` method public, and have callers explicitly enable/disable follow me by calling either method
* move the logic in `toggleFollowMeState(...)` to `enableFollowMe(...)` and/or `disableFollowMe()` (where appropriate), and remove the `toggleFollowMeState(...)` method.
* remove the `setLocationSource(...)` method, and pass the `FollowLocationSource` parameter to the `enableFollowMe(...)` method instead. In order to switch `FollowLocationSource`, the caller will have to call `enableFollowMe(...)` again with the updated `FollowLocationSource` parameter.